include/matx/operators/interp.h

```diff
@@ -0,0 +1,474 @@
+// BSD 3-Clause License
+//
+// Copyright (c) 2025, NVIDIA Corporation
+// All rights reserved.
+//
+// Redistribution and use in source and binary forms, with or without
+// modification, are permitted provided that the following conditions are met:
+//
+// 1. Redistributions of source code must retain the above copyright notice, this
+//    list of conditions and the following disclaimer.
+//
+// 2. Redistributions in binary form must reproduce the above copyright notice,
+//    this list of conditions and the following disclaimer in the documentation
+//    and/or other materials provided with the distribution.
+//
+// 3. Neither the name of the copyright holder nor the names of its
+//    contributors may be used to endorse or promote products derived from
+//    this software without specific prior written permission.
+//
+// THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+// AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+// IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+// DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+// FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+// DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+// SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+// CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+// OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+// OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+/////////////////////////////////////////////////////////////////////////////////
+
+#pragma once
+
+#include "matx/core/type_utils.h"
+#include "matx/operators/base_operator.h"
+
+namespace matx {
+
+  // Base interpolation method tag
+  struct InterpMethodBase {};
+
+  // Specific interpolation method tags
+  struct InterpMethodLinear : public InterpMethodBase {};
+  struct InterpMethodNearest : public InterpMethodBase {};
+  struct InterpMethodNext : public InterpMethodBase {};
+  struct InterpMethodPrev : public InterpMethodBase {};
+  struct InterpMethodSpline : public InterpMethodBase {};
+
+  namespace detail {
+    template <class O, class OpX, class OpV>
+    class InterpSplineTridiagonalFillOp : public BaseOp<InterpSplineTridiagonalFillOp<O, OpX, OpV>> {
+      // this is a custom operator that fills a tridiagonal system
+      // for cubic spline interpolation
+
+      using matxop = bool;
+    private:
+      O dl_, d_, du_, b_;
+      OpX x_;
+      OpV v_;
+
+    public:
+      InterpSplineTridiagonalFillOp(O dl, O d, O du, O b, OpX x, OpV v)
+          : dl_(dl), d_(d), du_(du), b_(b), x_(x), v_(v)  {}
+
+      __device__ inline void operator()(index_t idx) const
+      {
+      
+        if (idx == 0) { // left boundary condition
+          typename OpX::value_type x0 = x_(idx);
```

Comment:
May want to create aliases at the top for all these so it's not repeated. Something like:
```
using x_val_type = typename OpX::value_type;
```